### ISSUE TITLE
Resolve referenced json schema

### ISSUE THREAD

**Issue by hillin** [remcohaszing/monaco-yaml#107]
We have a large json schema file referencing some external json schema files using relative paths, e.g.

```json
{
    $ref: "../external/schema.json"
}
```

Currently this cannot be handled by monaco-yaml, it attempts to fetch these reference with the file protocol:
> Fetch API cannot load file:///../external/schema.json. URL scheme "file" is not supported.

It would be great if either
- externally json schema referenced with a relative path can be resolved correctly, or
- externally referenced json schema could be resolved with a custom resolver (maybe as an option passed to `setDiagnosticsOptions`).

**Comment by remcohaszing** [comment]
I’m note entirely sure if this can be resolved in `monaco-yaml` or if it needs to be resolved upstream in `yaml-language-server`. This needs some investigation.

I also ran into this when trying the schema on `https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/v1.20.5-standalone-strict/all.json` as part of investigating https://github.com/remcohaszing/monaco-yaml/issues/54#issuecomment-908653577

**Comment by Lyokolux** [comment]
Do you have any workarounds. I am new to the jsonschema and I am currently trying to understand if there is a way to use provided schema in monaco-yaml.

I am currently importing the schema, but it has a reference to another schema and this caused the validation to fails with the error:
```
Problems loading reference 'file:///name.schema.json#/definitions/Allocation': Unable to load schema from '/name.schema.json': Failed to fetch.
```

**Comment by hillin** [comment]
We have ditched this approach but I guess you can preprocess the schema file and replace those relative references with absolute URIs.

**Comment by ivy-rew** [comment]
Just started to use monaco-yaml and it seems to me as if this feature now works out of the box.

In my screenshot I have an inline schema pointing to https://json-schema.axonivy.com/app/0.0.1/app.json . This schema contains multiple child refs and they are all resolved.
![monaco-yaml-resolves](https://github.com/remcohaszing/monaco-yaml/assets/15085693/0624b83e-70c9-4e23-858b-c772f21ab10a)

Strange fact I noticed, with the latest available release 4.0.4 version of monaco-yaml referenced schemas are not resolved.
If I build the current 'main' branch manually as 4.0.5, using `npm pack` however, the feature works out of the box.
So I'd assume that this was fixed by accident. E.g. by consuming newer dependencies? :thinking:
